### PR TITLE
MORF-55 Add support for EXCEPT set operator and DB-link

### PR DESCRIPTION
**Pull request by michalradziwon**

### CODE REVIEW

File: morf-core/src/main/java/org/alfasoftware/morf/sql/element/TableReference.java [hunk 1]
```diff
@@ -52,6 +52,12 @@ public class TableReference implements DeepCopyableWithTransformation<TableRefer
    */
   private final boolean temporary;
 
+  /**
+   * If the table exists in another database then it may be possible to access it with an appropriate database link
+   * (not supported by some database platforms)
+   */
+  private final String dblink;
+
```
Comment (jsimlo) [inline morf-core/src/main/java/org/alfasoftware/morf/sql/element/TableReference.java:60]: Why are you initialising this with empty strings intead of nulls?

Note to self: No schemaName is denoted as null, but no alias is denoted as empty string.
So we have a precedence either way, I have not looked into why empty strings too much.

File: morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java [hunk 2]
```diff
@@ -745,15 +746,14 @@ protected String schemaNamePrefix(@SuppressWarnings("unused") Table table) {
```
Comment (jsimlo) [inline morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java:746]: Was it not possible to also drop all the other schemaNamePrefix() methods?

**Review by jsimlo — COMMENTED** [review]
Good stuff here.
And thanks for changing that schemaNamePrefix() approach.
Just two questions to wrap this up.

File: morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java (hunk 2 above)
Comment (michalradziwon) [reply morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java:746]: These calls can't be replaced with `tableNameWithSchemaName(TableReference tableRef)`:

```
    statement.append("INDEX ")
      .append(schemaNamePrefix(table))
      .append(index.getName())
```

```
  public Collection<String> renameIndexStatements(Table table, String fromIndexName, String toIndexName) {
    return ImmutableList.of("ALTER INDEX " + schemaNamePrefix(table) + fromIndexName + " RENAME TO " + toIndexName);
  }
```

Given the goal here was not to completely refactor the code I just left that as it is.

File: morf-core/src/main/java/org/alfasoftware/morf/sql/element/TableReference.java (hunk 1 above)
Comment (michalradziwon) [reply morf-core/src/main/java/org/alfasoftware/morf/sql/element/TableReference.java:60]: no specific reason for that :roll_eyes: do we want to change that to `null` ?

**Comment by michalradziwon** [comment]
thank you for reviewing Juraj!

**Review by jsimlo — APPROVED** [review]

**Comment by jsimlo** [comment]
@capgen628, this can be merged and released at @michalradziwon's convenience.

**Pull request merged**